extensions/amp-analytics/0.1/test/test-variables.js

```diff
@@ -193,6 +193,14 @@ describe('amp-analytics.VariableService', function() {
                 'S7Uc5ZmODduHWdplzrZ7Jsnqx')
           );
     });
+
+    it('replaces', () => {
+      return check('REPLACE(this-is-a-test, `-`, )', 'thisisatest');
```

Comment:
i thought this should gives `this is a test` (with spaces)

Reply:
we are currently stripping spaces so people can do things like `DEFAULT(one, two)` vs only allowing `DEFAULT(one,two)`. We could change this but I think allowing it is better?  It does make it tricky to pass in a space though

Reply:
Isn't that both tests are testing replace with no third arg? here `newSubStr = undefined`?

Reply:
no, `newSubStr === ''` in the first. I will add a few more tests to make this more clear

Reply:
so how do i get `this is a test`?  Is it supported?

Reply:
I was spending some time thinking about this last night. Right now it is not supported without using a different macro that would return a space. We could support something like ``REPLACE(this-is-a-test, `-`, ` `)`` If we do that should we make the back ticks on the third parameter required?  Also no matter what the spaces will be encodes so it would return `this%20is%20a%20test`

Reply:
We are now respecting whitespaced inside all back ticks. More tests were written to cover this behavior.